File: proyectos/src/main/java/utn/dds/persistencia/proyectos/model/ComplejidadMaxima.java
package utn.dds.persistencia.proyectos.model; 

import javax.persistence.*;

@Entity
public class ComplejidadMaxima extends ComplejidadMinima{

	@Override
	public double  costo(double tiempo){
		double costo = super.costo(tiempo);
		costo = costo * 1.07;
		if(tiempo > 10)
			return costo + (tiempo-10) * 10;
		else 
			return costo;
	}
}

File: manejo-project/src/ComplejidadMaxima.scala
class ComplejidadMaxima extends ComplejidadMinima{

  override def obtenerCosto(tiempo:Integer):Double ={
    if(tiempo<=10)
    	return (tiempo * 25)*1.07;
    else
      return (tiempo * 25)*1.07 + 10 *(tiempo-10);
  } 

}